contracts/finance/andromeda-weighted-distribution-splitter/src/contract.rs

```diff
@@ -0,0 +1,938 @@
+
+    // If the index exists, remove the element found in the index
+    // If the index doesn't exist, return an error
+    require(user_index != None, ContractError::UserNotFound {})?;
+
+    if let Some(i) = user_index {
+        splitter.recipients.swap_remove(i);
+        let new_splitter = Splitter {
+            recipients: splitter.recipients,
+            locked: splitter.locked,
+        };
+        SPLITTER.save(deps.storage, &new_splitter)?;
+    };
+
+    // // 2nd method: Filter out the recipient then create another splitter with the new recipients list
+    // let new_recipients: Vec<AddressWeight> = splitter
+    //     .recipients
+    //     .clone()
+    //     .into_iter()
+    //     .filter(|x| x.recipient != recipient)
+    //     .collect();
+    // // If the new vector has the same length as the old one, it means that the recipient wasn't found
+    // if new_recipients.len() == splitter.recipients.len() {
+    //     StdError::generic_err("User not found");
+    // }
+    // let new = Splitter {
+    //     recipients: new_recipients,
+    //     locked: splitter.locked,
+    // };
+
+    Ok(Response::default().add_attributes(vec![attr("action", "removed_recipient")]))
+}
+
+fn execute_update_lock(
+    deps: DepsMut,
+    info: MessageInfo,
+    lock: bool,
+) -> Result<Response, ContractError> {
+    require(
+        ADOContract::default().is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+
+    // No need to send funds
+    require(
+        info.funds.is_empty(),
+        ContractError::FunctionDeclinesFunds {},
+    )?;
+
+    let mut splitter = SPLITTER.load(deps.storage)?;
+    splitter.locked = lock;
+    SPLITTER.save(deps.storage, &splitter)?;
+
+    Ok(Response::default().add_attributes(vec![
+        attr("action", "update_lock"),
+        attr("locked", lock.to_string()),
+    ]))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
+    let version = get_contract_version(deps.storage)?;
+    if version.contract != CONTRACT_NAME {
+        return Err(ContractError::CannotMigrate {
+            previous_contract: version.contract,
+        });
+    }
+    Ok(Response::default())
+}
+
+#[entry_point]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<Binary, ContractError> {
+    match msg {
+        QueryMsg::GetSplitterConfig {} => encode_binary(&query_splitter(deps)?),
+        QueryMsg::GetUserWeight { user } => encode_binary(&query_user_weight(deps, user)?),
+        QueryMsg::AndrQuery(msg) => ADOContract::default().query(deps, env, msg, query),
+    }
+}
+
+fn query_user_weight(deps: Deps, user: Recipient) -> Result<GetUserWeightResponse, ContractError> {
+    let splitter = SPLITTER.load(deps.storage)?;
+    let recipients = splitter.recipients;
+
+    let addrs: Vec<AddressWeight> = recipients
+        .clone()
+        .into_iter()
```

Comment:
I think you can get away with an `iter` to avoid cloning.

Reply:
Also since you are enforcing uniqueness, `find` seems more appropriate: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find